x-pack/spec/monitoring/inputs/metrics/stats_event_factory_spec.rb

```diff
@@ -2,9 +2,13 @@
 # or more contributor license agreements. Licensed under the Elastic License;
 # you may not use this file except in compliance with the Elastic License.
 
+require "logstash/agent"
+require "logstash/runner"
+require "spec_helper"
```

Comment:
do you really need to add these "require" files to disable test cases?

Reply:
Yes and no. Without those, the single test run won't work, claiming it misses constant definitions for some classes, and to reproduce more easily the flaky test is useful to launch only that test and not the entire suite.